mautic/mautic
# Fix timezone issue on segment rebuild

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

At the moment relative date strings are not correctly use in segment filters.
We use two date types - date and datetime.  
Date should use regular timezone from configuration
Datetime should use UTC .

See some examples:

If now is 9. 1. 2023 6:29 am

Before:

- expression for date field and `today` string     `.. = 2023-01-09`
- expression datetime field and `today` string  `.. like 2023-01-09%`
- expression for date field and `next week` string  `.. = between 2023-01-09`

after 

- expression for date field and `today` string    `.. = 2023-01-09`
- expression datetime field and `today` string  `.. between 2023-01-08 23:00:00 and  2023-01-09 22:59:59 `

This PR is covered by unit tests



#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2.  Check If segments rebuild for date and DateTime fields works correctly
3. You can test filters like today, next week or relative dates like +3 days or -4 days
